client/src/modules/dashboard/Chapters/pages/ChaptersPage.tsx

```diff
@@ -38,9 +47,40 @@ const actionLinks = [
   },
 ];
 
+type searchState = {
+  search: string;
+};
+
+interface SearchAction {
+  type: 'setSearch';
+  payload: string;
+}
+
+const useFilter = () => {
+  const [{ search }, dispatch] = useReducer(
+    (state: searchState, action: SearchAction) => {
+      switch (action.type) {
+        case 'setSearch':
+          return { ...state, search: action.payload };
+      }
+    },
+    { search: '' },
+  );
+
+  const setSearch = useCallback((search: string) => {
+    dispatch({
+      type: 'setSearch',
+      payload: search,
+    });
+  }, []);
+
+  return { setSearch, search };
+};
```

Comment:
```suggestion
```
I think we could clean up the codebase by applying `useReducer` instead of juggling a bunch of `useState`s. However, in this case, it's a complicated way to mimic a single `useState`.

Reply:
Sorry, I somehow missed the entire discussion. I'll revisit this tomorrow, but for a quick answer: if you're planning to implement those things straight away, let's go with the reducer. If you're not, state is fine for now.

The rationale is just [YAGNI](https://en.m.wikipedia.org/wiki/You_aren't_gonna_need_it), so if you _do_ need it, go right ahead.

Reply:
It won't need to reduce an array state, if I don't include the tags, and it will need to reduce 2 boolean states and string.

I won't lie, I would like to delete it and use state to apply this https://github.com/freeCodeCamp/chapter/pull/2301#issuecomment-1408387596 and gain more experience, up to you tho, if this should stay or not, I am happy either way.